app/src/main/java/org/simple/clinic/summary/PatientSummaryScreen.kt

```diff
@@ -273,18 +276,13 @@ class PatientSummaryScreen :
     super.onRestoreInstanceState(mobiusDelegate.onRestoreInstanceState(state))
   }
 
-  @SuppressLint("CheckResult")
-  override fun onFinishInflate() {
-    super.onFinishInflate()
-
-    if (isInEditMode) {
-      return
-    }
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
 
     // Not sure why but the keyboard stays visible when coming from search.
     rootLayout.hideKeyboard()
 
-    val screenDestroys: Observable<ScreenDestroyed> = detaches().map { ScreenDestroyed() }
+    val screenDestroys: Observable<ScreenDestroyed> = view.detaches().map { ScreenDestroyed() }
```

Comment:
If we have situations like this when migrating the older screens, the right way to do it now is to manually track the subscriptions in a `CompositeDisposable` and manage them explicitly in the `onViewCreated()`/`onDestroyView()` lifecycle methods. Please see `InstantSearchScreen` for an example.

We can move this to the `BaseScreen` classes at some point, but for now this is what we need to do.

Reply:
I have done that in a later commit, where I traced `ScreenDestroyed` using `PublishSubject`.

Reply:
What I mean is that we should not be using `detaches()` anymore in a fragment based screen and instead should be using the `onViewCreated()` and `onDestroyView()` callbacks to manage subscriptions like this.

Reply:
Oh wait, I see what you mean. In any case, can we follow one method of maintaining these kind of subscriptions? I am planning to make this kind of thing easier to manage in the base class so doing it in one way would make it faster to migrate in the future.

Reply:
Yes, that would be great. We can expose observables from the base screen that gives the lifecycles events. That should make it easier when migrating since all we have to do is use the correct lifecycle variable. WDYT?

Reply:
We could think of that, yes. I was thinking of creating a `DelegateableFragment` at some point that allows us to delegate lifecycle method callbacks, and then extract common functionality from all our `BaseScreen` types into composable delegate types. This could be a good fit for that.

Reply:
Tracking here: https://app.clubhouse.io/simpledotorg/story/2646/implement-delegateablefragment-to-delegate-lifecycle-method-callbacks

Reply:
Didn't we make the change to remove the `screenDestroys`? I still see it in the latest change set 🤔

Reply:
I did. Well, I am no longer using detaches instead I am triggering a `PublishSubject` in `onDestroyView`
https://github.com/simpledotorg/simple-android/pull/2378/commits/cef1f65bcfebf74ad51684d893ca3927b9289b75

Reply:
I mean, can we follow a single approach to doing this sort of thing, like we did in `InstantSearchScreen`  by using a `CompositeDisposable` to track the subscriptions instead?